### PR TITLE
#397 : Implementing TeeReader

### PR DESCRIPTION
**Pull request by memoyil**
Implementing TeeReader

### CODE REVIEW

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#398` is in scope.

**Comment by 0crat** [comment]
+15 points just awarded to @memoyil, total is [+90](http://www.0crat.com/u/memoyil).

File: src/main/java/org/cactoos/io/ReaderOf.java [hunk 1]
```diff
@@ -278,6 +279,35 @@ public ReaderOf(final InputStream stream, final CharsetDecoder decoder) {
 
     /**
      * Ctor.
+     * @param input The input
+     * @param output The output
+     */
+    public ReaderOf(final Input input, final Output output) {
```
Comment (yegor256) [inline src/main/java/org/cactoos/io/ReaderOf.java]: @memoyil I don't think we need this. Just `TeeReader` is enough.

File: src/main/java/org/cactoos/io/TeeReader.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    // @checkstyle ParameterNumberCheck (8 line)
+    /**
+     * Ctor.
+     * @param input The source
+     * @param incharset The source charset
+     * @param out The destination
+     * @param outcharset The destination charset
+     */
+    TeeReader(final Input input, final Charset incharset,
```
Comment (yegor256) [inline src/main/java/org/cactoos/io/TeeReader.java]: @memoyil this ctor looks weird. Just `TeeReader(Reader, Writer)` is enough.

File: src/main/java/org/cactoos/io/TeeReader.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        final int done = this.reader.read(cbuf, 0, length);
+        if (done >= 0) {
+            this.writer.write(cbuf);
+        }
+        return done;
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.reader.close();
```
Comment (yegor256) [inline src/main/java/org/cactoos/io/TeeReader.java]: @memoyil if this line will throw an exception, the second line won't be executed. Right? It's bad.

File: src/test/java/org/cactoos/io/TeeReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ * @author Mehmet Yildirim (memoyil@gmail.com)
+ * @version $Id$
+ * @since 0.13
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class TeeReaderTest {
+
+    @Test
+    public void testTeeReader() throws IOException {
+        final Path src = Files.createTempFile("cactoos-1", "txt-1");
```
Comment (yegor256) [inline src/test/java/org/cactoos/io/TeeReaderTest.java]: @memoyil what's the point of using files? Can't we test with just in-memory data?

**Comment by yegor256** [comment]
@memoyil see above

File: src/main/java/org/cactoos/io/ReaderOf.java (hunk 1 above)
Comment (memoyil) [inline src/main/java/org/cactoos/io/ReaderOf.java]: Fixed

File: src/main/java/org/cactoos/io/TeeReader.java (hunk 2 above)
Comment (memoyil) [inline src/main/java/org/cactoos/io/TeeReader.java]: Fixed

File: src/main/java/org/cactoos/io/TeeReader.java (hunk 3 above)
Comment (memoyil) [inline src/main/java/org/cactoos/io/TeeReader.java]: Fixed

File: src/test/java/org/cactoos/io/TeeReaderTest.java (hunk 4 above)
Comment (memoyil) [inline src/test/java/org/cactoos/io/TeeReaderTest.java]: Fixed

**Comment by memoyil** [comment]
@yegor256 Fixed the code based on your comments

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#398` assigned to @kkamkou, please go ahead ([policy](http://datum.zerocracy.com/pages/policy.html)).

**Comment by 0crat** [comment]
@kkamkou resigned from `gh:yegor256/cactoos#398`, please stop working.

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#398` assigned to @kkamkou, please go ahead ([policy](http://datum.zerocracy.com/pages/policy.html)).

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @kkamkou, total is [+930](http://www.0crat.com/u/kkamkou).

**Comment by 0crat** [comment]
Oops! Job `gh:yegor256/cactoos#398` is not in scope

**Pull request merged**